src/events/reactionroles.ts

```diff
@@ -0,0 +1,210 @@
+        }
+
+        await member.roles.remove(
+            rolesArr,
+            "Reaction roles"
+        );
+
+        await member.roles.add(
+            role,
+            "Reaction roles"
+        );
+
+        await button.reply({
+            content: `Added <@&${role}>`,
+            ephemeral: true
+        });
+    } else {
+        // The confirm style only allows users to get a role
+        await member.roles.add(role);
+
+            await button.reply({
+                content: `Added <@&${role}>`,
+                ephemeral: true
+            });
+    }
+};
+
+/**
+ * Six hours in milliseconds. Used as interval for refreshing reaction roles.
+ */
+const SIX_HOURS: number = 21_600_000;
+
+let refreshInitiated: boolean = false;
+
+/**
+ * Refreshes all reaction roles in the database, deleting ones that no longer exist.
+ * 
+ * @param client the Discord bot
+ */
+const refreshReactionRoles = async (client: Client): Promise<void> => {
+    await getAdminLogsChannel().send("Refreshing reaction roles...");
+
+    // For each server...
+    for await (const [serverId, ] of client.guilds.cache) {
+        const server: Server = await Server.get(serverId);
+
+        const reactionMessages: ReadonlyMap<Snowflake, ReactionRoles> = server.reactionRoles;
+
+        // Skip servers without reaction roles
+        if (!reactionMessages) {
+            continue;
+        }
+
+        // Check all reaction roles
+        for await (const [messageId, reactionRoles] of reactionMessages) {
+            const channelId: Snowflake = reactionRoles.channel;
+
+            // Reaction role channels are always text channels
+            const channel: TextChannel = <TextChannel> client.channels.cache.get(channelId);
+
+            try {
+                // Check if the message exists
+                await channel.messages.fetch(messageId);
+            } catch {
+                // Delete the configuration if the message is gone
+                server.deleteReactionRoles(messageId);
+                await getAdminLogsChannel().send(
+                    `Deleted reaction roles for **${messageId}** in **${channel.guild.name}**`
+                );
+            }
+        }
+
+        // Save to the database
+        await server.save();
+    }
+
+    await getAdminLogsChannel().send("Reaction roles refreshed!");
+};
+
+/**
+ * Initiates an interval that refreshes the reaction roles data in the database.
+ *  
+ * @param client the Discord bot
+ */
+export const initRefreshReactionRolesInterval = async (client: Client): Promise<void> => {
+    // Do nothing if this has already been initiated
+    if (refreshInitiated) {
+        return;
+    }
+
+    // Otherwise initiate
+    refreshInitiated = true;
+
+    await refreshReactionRoles(client);
+
+    // Create a co-routine that triggers every six hours
+    setInterval(async (): Promise<void> => {
+        await refreshReactionRoles(client);
+    }, SIX_HOURS);
+};
```

Comment:
The stored reaction roles are set to refresh every six hours, to remove deleted reaction roles.